R/fit_SE.R

```diff
@@ -24,10 +26,8 @@ fit_SE <- function(se,
   checkmate::assert_number(pcutoff)
   checkmate::assert_number(cap)
   checkmate::assert_character(curve_type)
-  
-  req_assays <- c(averaged_assay)
-  lapply(req_assays, function(x) gDRutils::validate_se_assay_name(se, x))
-
+  checkmate::assert_true(all(curve_type %in% c("GR", "RV")))
+  gDRutils::validate_se_assay_name(se, averaged_assay)
```

Comment:
```suggestion
  gDRutils::validate_se_assay_name(se, metrics_assay)
```
These both are more strict than
```
  checkmate::assert_string(averaged_assay)
  checkmate::assert_string(metrics_assay)
```

Reply:
averaged_assay: I think it is better to leave both checkmate::assert_string(averaged_assay) and gDRutils::validate_se_assay_name(se, averaged_assay). First checks technical validity, second is for substantive evaluation of the argument.
metrics_assay: I'll add validation on that, thanks!

Reply:
✔️